src/main/webapp/courseLoader.js

```diff
@@ -271,11 +266,8 @@
   }
 
   function newURL(
```

Comment:
It strikes me that "newURL" isn't a very descriptive function name. Also, nit: It should be "newUrl"